core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -143,6 +143,14 @@ private TableProperties() {
   public static final String DELETE_ORC_BLOCK_SIZE_BYTES = "write.delete.orc.block-size-bytes";
   public static final long ORC_BLOCK_SIZE_BYTES_DEFAULT = 256L * 1024 * 1024; // 256 MB
 
+  public static final String ORC_COMPRESSION = "write.orc.compression-codec";
+  public static final String DELETE_ORC_COMPRESSION = "write.delete.orc.compression-codec";
+  public static final String ORC_COMPRESSION_DEFAULT = "zlib";
```

Comment:
Other formats use "gzip" for this. Is this what ORC requires?

Reply:
I think ORC as a format does not support GZIP. https://orc.apache.org/api/orc-core/org/apache/orc/CompressionKind.html

Engines like Presto/Trino tend to have ZLIB as a default compression mode for ORC.

Reply:
Given `ZSTD`'s good performance, are we considering it as the default?

Reply:
Although I believe ZSTD has been used in production by many users already, I personally prefer to align with the ORC default. Besides, I think we'd better start a new thread to discuss this.

Reply:
Yeah, I agree with @zhongyujiang, it's good to align with the apache orc default value. 
 According to the document https://orc.apache.org/docs/hive-config.html .